application/helpers/update/updates/Update_607.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace LimeSurvey\Helpers\Update;
+
+use Exception;
+
+/**
+ * Fix organizer link : icon and survey activated
+ * @package LimeSurvey\Helpers\Update
+ */
+class Update_607 extends DatabaseUpdateBase
+{
+    /**
+     * @inheritDoc
+     * @throws \CException
+     */
+    public function up()
+    {
+        $surveyThemes = $this->db->createCommand()
+            ->select('*')
+            ->from('{{templates}}')
+            ->queryAll();
+        if (!empty($surveyThemes)) {
+            foreach ($surveyThemes as $surveyTheme) {
+                $surveyThemeName = $surveyTheme['name'];
+                setTransactionBookmark();
+                try {
+                    if (!$this->isStandardTemplate($surveyThemeName)) {
+                        $rowsDeleted = $this->db->createCommand()->delete(
+                            '{{templates}}',
+                            'name=:name',
+                            [':name' => $surveyThemeName]
+                        );
+                        if ($rowsDeleted >= 1) {
+                            $this->db->createCommand()->delete(
+                                '{{template_configuration}}',
+                                'template_name=:templateName',
+                                [':templateName' => $surveyThemeName]
+                            );
+                        }
+                    }
+                } catch (Exception $e) {
+                    rollBackToTransactionBookmark();
```

Comment:
Will this work even if you have multiple bookmarks set? Or do we need to remove the bookmark on success?

Reply:
as far as i read up it always saves the newest savepoint when you generate one with the same name, so there shouldnt be any issues, correct me if im wrong. Changed the name to "beforeThemeDelete"